toto/util.py

```
@@ -97,6 +97,15 @@ def import_rsa_key_from_file(filepath, password=None):
 
   return rsa_key
 
+def import_rsa_public_keys_from_files_as_dict(filepaths):
+  """Takes a list of filepaths to RSA public keys and returns them as a
+  dictionary conformant with ssl_crypto.formats.KEYDICT_SCHEMA."""
+  key_dict = {}
+  for filepath in filepaths:
+    key = import_rsa_key_from_file(filepath)
+    keyid = key['keyid']
+    key_dict[keyid] = key
+  return key_dict
```

It may be a good idea to add statements to check whether the arguments of this function are sane (e.g., am I even getting a list?!)

---

Hm. I'm not so sure about it. If the the functions gets the wrong arguments it should raise an exception anyway. Maybe we want  something else than a `'TypeError' ... not iterable`?

---

>  Maybe we want something else than a 'TypeError' ... not iterable?

This is exactly what I mean. Is it a list, or an iterable? As of now, this function works with anything that implements `__iter__`, including a string. I don't know if this is intended, or even if it's a big deal, but it's something we should consider. We are offloading the sanity of our arguments to the for loop in there and `import_rsa_key_from_file`.

---

Yep. I think it is fine to just pipe through the exceptions from `import_rsa_key_from_file`. It will complain if the caller passes an `iterable` that does not contain file paths to keys.
We can use  a custom class based exception that is raised if the caller doesn't pass an `iterable`. But I'd prefer doing this here: https://github.com/in-toto/in-toto/issues/6#issuecomment-256983523